### DISCUSSION TITLE
Add questions or request features here.

### DISCUSSION THREAD

**Discussion by ToofDerling** [yegon666/CbzMage#1] [Announcements]
And please use issues for any bug reports :)

**Comment by gnamyl** [comment]
First of all this is awesome, thanks for coming through with this.  This fully replaces the old functionality IMO and in fact is better speedwise.  The entire Paper Girls compendium was seconds to generate the download.
So, let me present you with a scenario: I downloaded almost all my comixology stuff years ago with the old addon and i'm happy with those downloads, don't need to redownload them.  The problem with the command line is if I want the comics to be in kindle but also download something new I bought it's going to try and re-download something i don't need.  So I'd propose a future request :
Have some way to go in and "flag all comics present in the azw directory as already downloaded" so I can download something to Kindle that i've already downloaded previously but say I want to read in kindle or look at in kindle.  

Even better, if you just allowed an "interactivity" switch/flag for the command like /prompt or something so that the command would process and each item it found for download you could tap y/n.

**Reply by ToofDerling** [reply]
If I read you correctly you can use the title files and the scan mode to do something like the scenario you propose

Let's say you have added four new comics to your Kdl library, A, B, C, & D. You want to convert A & B to cbz files, but ignore C & D because you have downloaded them already.

Run CbzMage with the AzwScan parameter. It should list the four comics in green. Now go to the CbzMage title files directory. You should see title files for the four comics with a ".NEW" in front of the filename. 

Move the title files for C & D to the converted titles directory (it will look neater if you remove the ".NEW" from the moved files, but that's purely visual). 

Next time you run CbzMage, scan or convert, it will ignore C & D because they're "already converted"

**Reply by gnamyl** [reply]
Yes that will work.  I can use that.  I would still like an automated way for cbzmage to do it, but obviously that's just a request for you to consider!
THank you for replying so quickly!

**Reply by ToofDerling** [reply]
Now that v0.26 is out of the door I'd like to look at this again. But I'm not 100% sure how you want it to work.  

Maybe it's important to note that CbzMage doesn't download anything. It converts comic book files in your Kdl library on your computer. 

The workaround I proposed above is a bit clunky, but what about simply converting everything and then delete the cbz files you already have? After that CbzMage will ignore all the converted books (unless you delete til titlefile in the Converted Titles directory).

Flagging all books in the Kdl library as converted is doable and relatively easy to do. An actual interactive mode for new books in the library is also doable, but probably more difficult to do.

**Comment by zimdog1701** [comment]
Hi - I'm working my way through the process of installing and using this on my Mac:

1. .NET 6.0 Runtime and Ghostscript are installed
2. _CbzMage1.0_macOS_ resides in /Users/myusername/Applications
3. The Kdl data directory (known as "My Kindle Content" on the Mac) is now located in _/Users/myusername/_
4. I've entered "_/Users/myusername/_" as the AzwDir in _AzwConvertSettings.json_

However, when I run CbzMage (_~/Applications/CbzMage1.0_macOS/CbzMage AzwConvert_), I get the following error:

_**CbzMage fatal error. UnauthorizedAccessException: Access to the path '/Users/Titles' is denied.**_

I'm not sure where this Titles directory is supposed to be found, nor how to provide access. If it's under another name on the Mac, I'm assuming I would enter its path as TitlesDir in _AzwConvertSettings.json_ - yes?

Bottom line, I think I need some guidance from someone who has managed to use CbzMage on a Mac.
